include/saber/geometry/point.hpp

```diff
@@ -0,0 +1,140 @@
+#ifndef SABER_GEOMETRY_POINT_HPP
+#define SABER_GEOMETRY_POINT_HPP
+#pragma once
+
+namespace saber::geometry {
+
+// REVIEW mnfitz 15jun2024:
+// design issues involved with developing Point class
+// 1. Appropriate namespace
+// 2. Separate function declaration and definition vs java style combined
+// 3. Has-a: naked types or std::tuple or std::array
+
+template<typename T>
+class Point
```

Comment:
Need more comments...
Every method in this class should have a Doxygen comment.
create a JIRA task for adding comments